mitosheet/mitosheet/plugins/README.md

```diff
@@ -0,0 +1,169 @@
+DeleteRowStepPerformer - Yes.
+PromoteRowToHeaderStepPerformer - Yes.
+TransposeStepPerformer - Yes.
+MeltStepPerformer - Yes.
+
+#### Yes, but Much Worse
+AddColumnStepPerformer - Yes.
+ReorderColumnStepPerformer - Yes? By index. Would be terrible.
+SortStepPerformer - Yes. Select would be a bit weird.
+SetCellValueStepPerformer - Yes. Would be kinda weird.
+SetColumnFormulaStepPerformer - Yes. Not the best interface
+SplitTextToColumnsStepPerformer - Yes (without a preview). So No.
+
+#### Partially Possible
+FilterStepPerformer - Yes. But would also be terrible.
+MergeStepPerformer - old version, yes.
+
+#### Not Possible
+PivotStepPerformer - No
+ChangeColumnFormatStepPerformer - Yes. But don't have access to this metadata, so No.
+ConcatStepPerformer - no.
+SimpleImportStepPerformer - No. Missing access to data from the file system
+ExcelImportStepPerformer - No. Missing access to data from the file system.
+GraphStepPerformer - No
+GraphDeleteStepPerformer - No
+GraphDuplicateStepPerformer - No
+GraphRenameStepPerformer - No
+
+
+## What does Bamboolib Do?
+
+See their examples here: https://github.com/tkrabel/bamboolib/tree/master/plugins
+
+Pretty much, they expose LoaderPlugin, TransformationPlugin, and allow you to override methods.
+
+Within these components, they let you use ipywidgets (and some select wrappers they have) to define specific HMTL elements. This allows you to
+
+
+
+## Syntax for Params, V1, attempt 1
+
+If we wanted to use some JSON syntax (this could allow us to keep things very constrained), then here's an idea of how this syntax could look:
+
+```
+{
+    'parameter_specification_version': 1,
+    'parameters': [
+        {
+            'param_name': 'sheet_index',
+            'type': 'DataframeSelect',
+            'display_name': 'Dataframe',
+            'description': 'Dataframe to FillNaN values in.'
+        },
+        {
+            'param_name': 'column_headers',
+            'type': 'ColumnMultiSelect',
+            'display_name': 'Columns to fill in',
+            'description': 'Dataframe to FillNaN values in.'
+        }
+        ...
+    ]
+}
+```
+
+This is indeed extremly constrained (far more constrained than Bamboolib). It would be possible to express the these step performers listed above, but not all of them, and not in the nicest ways.
+
+We should make some components (that we can use in places) for these things. So note that this isn't just the easiest thing in the world.
+
+### How to go from this JSON -> a Frontend
+
+It would be fairly easy to take this JSON, load it up into AnalysisData, and send it to the frontend. Then, we just loop over the params, and turn them into elements. There's nothing particularly complex about this.
+
+## Syntax for Params, V1, attempt 2
+
+We follow bamboolib's example and instead provide Python classes that allow you to build the plugins. I'm less sure how this would work, but you imagine returning a list of params... so imagine something like:
```

Comment:
It would be very cool to find some other people who have written Bamboolib plugins (maybe just through GitHub?) and ask them about their experience using that infrastructure.

Reply:
Cool idea -- I can't find anyone through Github though (they seem like they would generally be private) (https://github.com/search?l=&q=TransformationPlugin+language%3APython&type=code)